### PR TITLE
Tie dynarmic to ARCHITECTURE_x86_64

### PR DESCRIPTION
**Pull request by linkmauve**
This allows Citra to build on other architectures, such as ARM.

The JIT is disabled with a warning, telemetry doesn’t send any specific CPU information, but the emulator is perfectly usable.

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/3352)

### CODE REVIEW

**Comment by neobrain** [comment]
Hi, this is neobot, using neobrain's account. I'm keeping an archive of versions of this PR:

2018-01-08T17:03:35Z: https://github.com/citra-emu/citra/compare/a66e4585a0e69cad572332a3f9c267b5cdb3424b...linkmauve:e0405887bc9c8b44a5f9026d5f247074f1c0e6f9

File: src/core/CMakeLists.txt [hunk 1]
```diff
@@ -404,7 +400,17 @@ add_library(core STATIC
...
 
 target_link_libraries(core PUBLIC common PRIVATE audio_core network video_core)
-target_link_libraries(core PUBLIC Boost::boost PRIVATE cryptopp dynarmic fmt)
+target_link_libraries(core PUBLIC Boost::boost PRIVATE cryptopp fmt)
 if (ENABLE_WEB_SERVICE)
     target_link_libraries(core PUBLIC json-headers web_service)
 endif()
+
+if (ARCHITECTURE_x86_64)
+    add_library(core STATIC
```
Comment (MerryMage) [inline src/core/CMakeLists.txt]: s/`add_library(core STATIC`/`target_sources(core PRIVATE`/

**Comment by neobrain** [comment]
Hi, this is neobot, using neobrain's account. I'm keeping an archive of versions of this PR:

2018-01-08T18:14:08Z: https://github.com/citra-emu/citra/compare/a66e4585a0e69cad572332a3f9c267b5cdb3424b...linkmauve:216df3c1634253da9c681feff4f09aeb3ad90b89

**Comment by linkmauve** [comment]
@jroweboy, I will probably add some ARM-specific capabilities next, once Citra actually boots with the interpreter there.

Anyway I just tested on an AArch64 system, Citra fully builds with this PR (but doesn’t run yet) so I consider that a success. :)

**Comment by neobrain** [comment]
Hi, this is neobot, using neobrain's account. I'm keeping an archive of versions of this PR:

2018-01-08T18:21:29Z: https://github.com/citra-emu/citra/compare/a66e4585a0e69cad572332a3f9c267b5cdb3424b...linkmauve:7f9fae5f7c2d4d4e4ad385fc48c397d8761fd13a

**Pull request merged**